diff --git a/src/plugins/projectexplorer/appoutputpane.cpp b/src/plugins/projectexplorer/appoutputpane.cpp
index 9bea0e0d..60c04f86 100644
--- a/src/plugins/projectexplorer/appoutputpane.cpp
+++ b/src/plugins/projectexplorer/appoutputpane.cpp
@@ -160,6 +160,8 @@ AppOutputPane::AppOutputPane() :
 
     setObjectName("AppOutputPane"); // Used in valgrind engine
     loadSettings();
+    m_tabWidget->setStyleSheet("QTabBar::tab { height: 24px; }");
+    m_tabWidget->setTabBarAutoHide(true);
 
     // Rerun
     m_reRunButton->setIcon(Utils::Icons::RUN_SMALL_TOOLBAR.icon());
