diff -cr a/src/Parsers/CppParser/cppdocumentparser.cpp b/src/Parsers/CppParser/cppdocumentparser.cpp
*** a/src/Parsers/CppParser/cppdocumentparser.cpp	2022-07-07 19:34:42.332192580 +0300
--- b/src/Parsers/CppParser/cppdocumentparser.cpp	2022-07-07 19:33:42.658854850 +0300
***************
*** 41,47 ****
  #include <texteditor/syntaxhighlighter.h>
  #include <texteditor/texteditor.h>
  #include <utils/algorithm.h>
! #include <utils/mimetypes/mimedatabase.h>
  #include <utils/runextensions.h>
  
  #include <QApplication>
--- 41,48 ----
  #include <texteditor/syntaxhighlighter.h>
  #include <texteditor/texteditor.h>
  #include <utils/algorithm.h>
! #include <utils/mimetypes2/mimedatabase.h>
! #include <utils/mimeutils.h>
  #include <utils/runextensions.h>
  
  #include <QApplication>
