diff --git a/src/Parsers/CppParser/cppdocumentparser.cpp b/src/Parsers/CppParser/cppdocumentparser.cpp
index c95740d8..7edab6d0 100644
--- a/src/Parsers/CppParser/cppdocumentparser.cpp
+++ b/src/Parsers/CppParser/cppdocumentparser.cpp
@@ -43,6 +43,7 @@
 #include <utils/algorithm.h>
 #include <utils/mimeutils.h>
 #include <utils/runextensions.h>
+#include <utils/qtcassert.h>
 
 #include <QApplication>
 #include <QFutureWatcher>
diff --git a/src/Parsers/CppParser/cppdocumentprocessor.cpp b/src/Parsers/CppParser/cppdocumentprocessor.cpp
index 348fe143..b9c83d9a 100644
--- a/src/Parsers/CppParser/cppdocumentprocessor.cpp
+++ b/src/Parsers/CppParser/cppdocumentprocessor.cpp
@@ -25,6 +25,7 @@
 #include <cplusplus/Overview.h>
 #include <cppeditor/cppeditordocument.h>
 #include <cppeditor/cppdoxygen.h>
+#include <cppeditor/cppmodelmanager.h>
 
 using namespace SpellChecker;
 using namespace SpellChecker::CppSpellChecker::Internal;
