diff --git a/src/plugins/python/pythonproject.cpp b/src/plugins/python/pythonproject.cpp
index 036d5e7c..93a93188 100644
--- a/src/plugins/python/pythonproject.cpp
+++ b/src/plugins/python/pythonproject.cpp
@@ -219,12 +219,12 @@ void PythonBuildSystem::triggerParsing()
     auto newRoot = std::make_unique<PythonProjectNode>(projectDirectory());
 
     const FilePath projectFile = projectFilePath();
-    const QString displayName = projectFile.relativePathFrom(projectDirectory()).toUserOutput();
+    const QString displayName = projectFile.fileName();
     newRoot->addNestedNode(
         std::make_unique<PythonFileNode>(projectFile, displayName, FileType::Project));
 
     for (const FileEntry &entry : std::as_const(m_files)) {
-        const QString displayName = entry.filePath.relativePathFrom(projectDirectory()).toUserOutput();
+        const QString displayName = entry.filePath.fileName();
         const FileType fileType = getFileType(entry.filePath);
 
         newRoot->addNestedNode(std::make_unique<PythonFileNode>(entry.filePath, displayName, fileType));
