diff --git a/src/libs/utils/smallstring.h b/src/libs/utils/smallstring.h
index f4cd6b74..d93066dc 100644
--- a/src/libs/utils/smallstring.h
+++ b/src/libs/utils/smallstring.h
@@ -595,16 +595,7 @@ public:
     static
     BasicSmallString number(long long int number)
     {
-#ifdef __cpp_lib_to_chars
-        // +1 for the sign, +1 for the extra digit
-        char buffer[std::numeric_limits<long long int>::digits10 + 2];
-        auto result = std::to_chars(buffer, buffer + sizeof(buffer), number, 10);
-        Q_ASSERT(result.ec == std::errc{});
-        auto endOfConversionString = result.ptr;
-        return BasicSmallString(buffer, endOfConversionString);
-#else
         return std::to_string(number);
-#endif
     }
 
     static
