diff --git a/src/plugins/python/pythoneditor.cpp b/src/plugins/python/pythoneditor.cpp
index ca8ecff0..90011250 100644
--- a/src/plugins/python/pythoneditor.cpp
+++ b/src/plugins/python/pythoneditor.cpp
@@ -296,8 +296,7 @@ PythonDocument::PythonDocument()
 
 void PythonDocument::updateCurrentPython()
 {
-    if (!isTemporary())
-        updatePython(detectPython(filePath()));
+    updatePython(detectPython(filePath()));
 }
 
 void PythonDocument::updatePython(const FilePath &python)
diff --git a/src/plugins/python/pythonlanguageclient.cpp b/src/plugins/python/pythonlanguageclient.cpp
index e074a446..b11bdaaf 100644
--- a/src/plugins/python/pythonlanguageclient.cpp
+++ b/src/plugins/python/pythonlanguageclient.cpp
@@ -357,7 +360,7 @@ void PyLSConfigureAssistant::installPythonLanguageServer(const FilePath &python,
 void PyLSConfigureAssistant::openDocument(const FilePath &python, TextEditor::TextDocument *document)
 {
     resetEditorInfoBar(document);
-    if (!PythonSettings::pylsEnabled() || !python.exists() || document->isTemporary())
+    if (!PythonSettings::pylsEnabled() || !python.exists())
         return;
 
     if (auto client = pythonClients().value(python)) {
