diff -cr a/charproc.c b/charproc.c
*** a/charproc.c	2024-12-01 22:34:23.000000000 +0300
--- b/charproc.c	2024-12-17 21:26:21.358700292 +0300
***************
*** 9472,9478 ****
  	    if (rows < 0)
  		value = (unsigned long) MaxRows(screen);
  	    value *= (unsigned long) FontHeight(screen);
! 	    value += (unsigned long) (2 * screen->border);
  	    if (!okDimension(value, askedHeight))
  		goto give_up;
  	}
--- 9472,9478 ----
  	    if (rows < 0)
  		value = (unsigned long) MaxRows(screen);
  	    value *= (unsigned long) FontHeight(screen);
! 	    value += (unsigned long) (4 * screen->border);
  	    if (!okDimension(value, askedHeight))
  		goto give_up;
  	}
***************
*** 9481,9487 ****
  	    if (cols < 0)
  		value = (unsigned long) MaxCols(screen);
  	    value *= (unsigned long) FontWidth(screen);
! 	    value += (unsigned long) ((2 * screen->border)
  				      + ScrollbarWidth(screen));
  	    if (!okDimension(value, askedWidth))
  		goto give_up;
--- 9481,9487 ----
  	    if (cols < 0)
  		value = (unsigned long) MaxCols(screen);
  	    value *= (unsigned long) FontWidth(screen);
! 	    value += (unsigned long) ((4 * screen->border)
  				      + ScrollbarWidth(screen));
  	    if (!okDimension(value, askedWidth))
  		goto give_up;
diff -cr a/fontutils.c b/fontutils.c
*** a/fontutils.c	2024-12-01 23:29:42.000000000 +0300
--- b/fontutils.c	2024-12-17 21:26:21.358700292 +0300
***************
*** 3793,3799 ****
  	win->f_ascent = font->ascent;
  	win->f_descent = font->descent;
      }
!     i = 2 * screen->border + sbwidth;
      j = 2 * screen->border;
      width = MaxCols(screen) * win->f_width + i;
      height = MaxRows(screen) * win->f_height + j;
--- 3793,3799 ----
  	win->f_ascent = font->ascent;
  	win->f_descent = font->descent;
      }
!     i = 4 * screen->border + sbwidth;
      j = 2 * screen->border;
      width = MaxCols(screen) * win->f_width + i;
      height = MaxRows(screen) * win->f_height + j;
diff -cr a/ptyx.h b/ptyx.h
*** a/ptyx.h	2024-12-01 22:28:23.000000000 +0300
--- b/ptyx.h	2024-12-17 21:26:21.358700292 +0300
***************
*** 3679,3687 ****
  #define PrinterOf(screen) (screen)->printer_state
  
  #ifdef SCROLLBAR_RIGHT
! #define OriginX(screen) (((term->misc.useRight)?0:ScrollbarWidth(screen)) + screen->border)
  #else
! #define OriginX(screen) (ScrollbarWidth(screen) + screen->border)
  #endif
  
  #define OriginY(screen) (screen->border)
--- 3679,3687 ----
  #define PrinterOf(screen) (screen)->printer_state
  
  #ifdef SCROLLBAR_RIGHT
! #define OriginX(screen) (((term->misc.useRight)?0:ScrollbarWidth(screen)) + screen->border*2)
  #else
! #define OriginX(screen) (ScrollbarWidth(screen) + screen->border*2)
  #endif
  
  #define OriginY(screen) (screen->border)
diff -cr a/screen.c b/screen.c
*** a/screen.c	2024-12-01 23:14:26.000000000 +0300
--- b/screen.c	2024-12-17 21:26:21.358700292 +0300
***************
*** 1450,1456 ****
  		       VDrawable(screen),
  		       MakeMarkGC(xw, set),
  		       x, y,
! 		       (unsigned) screen->border,
  		       (unsigned) FontHeight(screen));
      }
  }
--- 1450,1456 ----
  		       VDrawable(screen),
  		       MakeMarkGC(xw, set),
  		       x, y,
! 		       (unsigned) 2*screen->border,
  		       (unsigned) FontHeight(screen));
      }
  }
***************
*** 2025,2031 ****
  	   (double) (width - border - ScrollbarWidth(screen)) / FontWidth(screen)));
  
      rows = (height - border) / FontHeight(screen);
!     cols = (width - border - ScrollbarWidth(screen)) / FontWidth(screen);
      if (rows < 1)
  	rows = 1;
      if (cols < 1)
--- 2025,2031 ----
  	   (double) (width - border - ScrollbarWidth(screen)) / FontWidth(screen)));
  
      rows = (height - border) / FontHeight(screen);
!     cols = (width - 2*border - ScrollbarWidth(screen)) / FontWidth(screen);
      if (rows < 1)
  	rows = 1;
      if (cols < 1)
***************
*** 2244,2250 ****
  	    set_cur_col(screen, screen->max_col);
  
  	screen->fullVwin.height = height - border;
! 	screen->fullVwin.width = width - border - screen->fullVwin.sb_info.width;
  
  	scroll_displayed_graphics(xw, -move_down_by);
      } else if (FullHeight(screen) == height && FullWidth(screen) == width) {
--- 2244,2250 ----
  	    set_cur_col(screen, screen->max_col);
  
  	screen->fullVwin.height = height - border;
! 	screen->fullVwin.width = width - 2*border - screen->fullVwin.sb_info.width;
  
  	scroll_displayed_graphics(xw, -move_down_by);
      } else if (FullHeight(screen) == height && FullWidth(screen) == width) {
***************
*** 3119,3125 ****
  set_resize_increments(XtermWidget xw)
  {
      TScreen *screen = TScreenOf(xw);
!     int min_width = (2 * screen->border) + screen->fullVwin.sb_info.width;
      int min_height = (2 * screen->border);
      XSizeHints sizehints;
  
--- 3119,3125 ----
  set_resize_increments(XtermWidget xw)
  {
      TScreen *screen = TScreenOf(xw);
!     int min_width = (4 * screen->border) + screen->fullVwin.sb_info.width;
      int min_height = (2 * screen->border);
      XSizeHints sizehints;
  
diff -cr a/scrollbar.c b/scrollbar.c
*** a/scrollbar.c	2024-12-01 23:27:00.000000000 +0300
--- b/scrollbar.c	2024-12-17 21:26:21.358700292 +0300
***************
*** 103,109 ****
      TScreen *screen = TScreenOf(xw);
  
      int border = 2 * screen->border;
!     int min_wide = border + screen->fullVwin.sb_info.width;
      int min_high = border;
      XtGeometryResult geomreqresult;
      Dimension reqWidth, reqHeight, repWidth, repHeight;
--- 103,109 ----
      TScreen *screen = TScreenOf(xw);
  
      int border = 2 * screen->border;
!     int min_wide = border*2 + screen->fullVwin.sb_info.width;
      int min_high = border;
      XtGeometryResult geomreqresult;
      Dimension reqWidth, reqHeight, repWidth, repHeight;
diff -cr a/util.c b/util.c
*** a/util.c	2024-12-01 23:14:26.000000000 +0300
--- b/util.c	2024-12-17 21:27:10.812033855 +0300
***************
*** 2388,2394 ****
      GC gc;
  
      if ((gc = vwin->border_gc) != NULL) {
! 	int vmark1 = screen->border;
  	int vmark2 = vwin->height + vmark1;
  	int hmark1 = OriginX(screen);
  	int hmark2 = vwin->width + hmark1;
--- 2388,2394 ----
      GC gc;
  
      if ((gc = vwin->border_gc) != NULL) {
! 	int vmark1 = screen->border * 2;
  	int vmark2 = vwin->height + vmark1;
  	int hmark1 = OriginX(screen);
  	int hmark2 = vwin->width + hmark1;
***************
*** 4919,4925 ****
      TRACE(("   border    %d\n", xw->core.border_width));
      TRACE(("   scrollbar %d\n", scrollbarWidth));
  
!     xw->hints.base_width = 2 * screen->border + scrollbarWidth;
      xw->hints.base_height = 2 * screen->border;
  
  #if OPT_TOOLBAR
--- 4919,4925 ----
      TRACE(("   border    %d\n", xw->core.border_width));
      TRACE(("   scrollbar %d\n", scrollbarWidth));
  
!     xw->hints.base_width = 4 * screen->border + scrollbarWidth;
      xw->hints.base_height = 2 * screen->border;
  
  #if OPT_TOOLBAR
