*** clementine-0.6-roand/src/ui/qtsystemtrayicon.cpp	2010-12-24 00:18:40.000000000 +0000
--- clementine-0.6/src/ui/qtsystemtrayicon.cpp	2010-12-11 17:30:37.000000000 +0000
***************
*** 231,236 ****
--- 231,237 ----
  
  void QtSystemTrayIcon::SetNowPlaying(const Song& song,
                                       const QString& image_path) {
+ return;
  #ifdef Q_OS_WIN
    // Windows doesn't support HTML in tooltips, so just show something basic
    tray_->setToolTip(song.PrettyTitleWithArtist());
