diff -cr old/src/border.c new/src/border.c
*** old/src/border.c	2024-10-09 23:22:26.036674030 +0300
--- new/src/border.c	2024-10-09 23:29:12.676680367 +0300
***************
*** 488,494 ****
              titley += south - 1;
           }
           RenderString(canvas, FONT_BORDER, borderTextColor,
!                       titlex, titley, titleWidth, titleBuffer);
           Release(titleBuffer);
        }
  
--- 488,494 ----
              titley += south - 1;
           }
           RenderString(canvas, FONT_BORDER, borderTextColor,
!                       titlex, titley + 1, titleWidth, titleBuffer);
           Release(titleBuffer);
        }
  
***************
*** 1168,1176 ****
  #ifdef USE_ICONS
     const char hasIcon = np->icon ? 1 : 0;
     const int titleHeight = GetTitleHeight();
!    const int iconSize = hasIcon ? GetBorderIconSize() : Max((int)titleHeight - 2, 0);
!    const int iconXOffset = (titleHeight - iconSize) / 2;
!    const int iconYOffset = hasIcon ? iconXOffset : iconXOffset + 1;
     IconNode *icon;
     if(hasIcon) {
        icon = np->icon;
--- 1168,1176 ----
  #ifdef USE_ICONS
     const char hasIcon = np->icon ? 1 : 0;
     const int titleHeight = GetTitleHeight();
!    const int iconSize = hasIcon ? GetBorderIconSize() + 2 : Max((int)titleHeight - 2, 0);
!    const int iconXOffset = (titleHeight - iconSize - 2) / 2 ;
!    const int iconYOffset = hasIcon ? (titleHeight - iconSize) / 2 : iconXOffset + 1;
     IconNode *icon;
     if(hasIcon) {
        icon = np->icon;
