diff --git a/src/root.c b/src/root.c
index b381982e..204c06c1 100644
--- a/src/root.c
+++ b/src/root.c
@@ -193,6 +193,7 @@ char ShowRootMenu(int index, int x, int y, char keyboard)
       menu = ParseDynamicMenu(menu->timeout_ms, menu->dynamic);
       if(menu) {
          InitializeMenu(menu);
+         menu->itemHeight = rootMenu[index]->itemHeight;
          ShowMenu(menu, RunRootCommand, x, y, keyboard);
          DestroyMenu(menu);
          return 1;

