diff --git a/src/root.c b/src/root.c
index b381982e..5df85ac3 100644
--- a/src/root.c
+++ b/src/root.c
@@ -192,6 +192,7 @@ char ShowRootMenu(int index, int x, int y, char keyboard)
       Menu *menu = rootMenu[index];
       menu = ParseDynamicMenu(menu->timeout_ms, menu->dynamic);
       if(menu) {
+         menu->itemHeight = 14;
          InitializeMenu(menu);
          ShowMenu(menu, RunRootCommand, x, y, keyboard);
          DestroyMenu(menu);
