diff -cr a/configure.ac b/configure.ac
*** a/configure.ac        2017-08-13 05:03:02.000000000 +0300
--- b/configure.ac        2017-08-13 12:54:28.736731758 +0300
***************
*** 6,8 ****
--- 6,9 ----
+ AM_INIT_AUTOMAKE
  AC_PREREQ(2.57)
  AC_CONFIG_SRCDIR([src])
  AC_CONFIG_SRCDIR([po])
***************
*** 499,505 ****
  ############################################################################
  AM_ICONV
  AM_GNU_GETTEXT([external])
! AM_GNU_GETTEXT_VERSION([0.19])
  LDFLAGS="$LDFLAGS $LIBINTL $LIBICONV"
  
  ############################################################################
--- 500,506 ----
  ############################################################################
  AM_ICONV
  AM_GNU_GETTEXT([external])
! AM_GNU_GETTEXT_VERSION([0.20])
  LDFLAGS="$LDFLAGS $LIBINTL $LIBICONV"
  
  ############################################################################
