diff -cr jwm-1271/src/taskbar.c jwm-roand/src/taskbar.c
*** jwm-1271/src/taskbar.c	2015-09-12 02:46:52.000000000 +0300
--- jwm-roand/src/taskbar.c	2015-09-18 20:06:44.000000000 +0300
***************
*** 755,766 ****
     ResetButton(&button, bp->cp->pixmap);
     button.border = settings.trayDecorations == DECO_MOTIF;
     button.font = FONT_TASKLIST;
!    button.height = bp->itemHeight;
!    button.width = bp->itemWidth;
     button.text = NULL;
  
!    x = 0;
!    y = 0;
     for(tp = taskEntries; tp; tp = tp->next) {
  
        if(!ShouldShowEntry(tp)) {
--- 755,766 ----
     ResetButton(&button, bp->cp->pixmap);
     button.border = settings.trayDecorations == DECO_MOTIF;
     button.font = FONT_TASKLIST;
!    button.height = bp->itemHeight - 2;
!    button.width = bp->itemWidth - 2;
     button.text = NULL;
  
!    x = 1;
!    y = 1;
     for(tp = taskEntries; tp; tp = tp->next) {
  
        if(!ShouldShowEntry(tp)) {
diff -cr jwm-1271/src/traybutton.c jwm-roand/src/traybutton.c
*** jwm-1271/src/traybutton.c	2015-09-12 02:46:52.000000000 +0300
--- jwm-roand/src/traybutton.c	2015-09-18 20:06:44.000000000 +0300
***************
*** 267,277 ****
     } else {
        button.type = BUTTON_TRAY;
     }
!    button.width = cp->width;
!    button.height = cp->height;
!    button.border = settings.trayDecorations == DECO_MOTIF;
!    button.x = 0;
!    button.y = 0;
     button.font = FONT_TRAY;
     button.text = bp->label;
     button.icon = bp->icon;
--- 267,280 ----
     } else {
        button.type = BUTTON_TRAY;
     }
!    button.width = cp->width - 2;
!    button.height = cp->height - 2;
!    if (bp->label || button.type == BUTTON_TRAY_ACTIVE)
!       button.border = settings.trayDecorations == DECO_MOTIF;
!    else
!       button.border=0;
!    button.x = 1;
!    button.y = 1;
     button.font = FONT_TRAY;
     button.text = bp->label;
     button.icon = bp->icon;
diff -cr jwm-1271/src/button.c jwm-roand/src/button.c
*** jwm-1271/src/button.c        2015-09-12 02:46:52.000000000 +0300
--- jwm-roand/src/button.c    2015-09-18 21:03:16.000000000 +0300
***************
*** 150,155 ****
--- 150,157 ----
        textHeight = GetStringHeight(bp->font);
        if(iconWidth > 0 && textWidth + iconWidth + 7 > width) {
           textWidth = width - iconWidth - 7;
+          if (bp->type==BUTTON_TASK || bp->type==BUTTON_TASK_ACTIVE)
+              textWidth = textWidth - 7; 
        } else if(iconWidth == 0 && textWidth + 5 > width) {
           textWidth = width - 5;
        }
