diff --git a/src/action.h b/src/action.h
index ce02359e..7bf4d0f5 100644
--- a/src/action.h
+++ b/src/action.h
@@ -65,6 +65,7 @@ typedef struct {
 #define ACTION_RESTORE        42
 #define ACTION_CENTER         43
 #define ACTION_AT             44
+#define ACTION_KILL           45
 #define ACTION_INVALID        255
 #define ACTION_RESIZE_N       1  /* Extra value mask for resize north. */
 #define ACTION_RESIZE_S       2  /* Extra value mask for resize south. */
diff --git a/src/binding.c b/src/binding.c
index 5b740820..6d2603e2 100644
--- a/src/binding.c
+++ b/src/binding.c
@@ -307,6 +307,7 @@ char ShouldGrab(ActionType action)
    case ACTION_RESTORE:
    case ACTION_CENTER:
    case ACTION_AT:
+   case ACTION_KILL:
       return 1;
    default:
       return 0;
diff --git a/src/event.c b/src/event.c
index 78e3ec55..211af756 100644
--- a/src/event.c
+++ b/src/event.c
@@ -709,6 +709,10 @@ void ProcessBinding(MouseContextType context, ClientNode *np,
       StartWindowWalk();
       FocusAt(key.extra);
       break;
+   case ACTION_KILL:
+      KillClient(np);
+      break;
+
    default:
       break;
    }
diff --git a/src/parse.c b/src/parse.c
index 6cb10ad6..000b10c7 100644
--- a/src/parse.c
+++ b/src/parse.c
@@ -52,6 +52,7 @@ static const StringMappingType ACTION_MAP[] = {
    { "escape",                ACTION_ESC           },
    { "exit",                  ACTION_EXIT          },
    { "fullscreen",            ACTION_FULLSCREEN    },
+   { "kill",                  ACTION_KILL          },
    { "ldesktop",              ACTION_LDESKTOP      },
    { "left",                  ACTION_LEFT          },
    { "maxbottom",             ACTION_MAXBOTTOM     },
