--- a/src/checks/manuallevel/qvariant-template-instantiation.cpp	2025-05-26 21:09:35.511857754 +0300
+++ b/src/checks/manuallevel/qvariant-template-instantiation.cpp	2025-04-27 21:16:40.000000000 +0300
@@ -57,7 +57,7 @@
         return;
     }
 
-    const auto *decl = dyn_cast<CXXRecordDecl>(memberExpr->getBase()->getType()->getAsCXXRecordDecl());
+    const auto *decl = memberExpr->getBase()->getType()->getAsCXXRecordDecl();
     if (!decl || !decl->getDefinition()
         || !(decl->getNameAsString() == "QVariant" || decl->getNameAsString() == "QHash" || decl->getNameAsString() == "QMap")) {
         return;
