--- a/src/checks/manuallevel/qvariant-template-instantiation.cpp	2025-05-14 10:27:16.021569159 +0300
+++ b/src/checks/manuallevel/qvariant-template-instantiation.cpp	2025-05-14 10:26:25.366273055 +0300
@@ -57,7 +57,13 @@
         return;
     }
 
-    const auto *decl = dyn_cast<CXXRecordDecl>(memberExpr->getBase()->getType()->getAsCXXRecordDecl());
+    auto *recordDecl = memberExpr->getBase()->getType()->getAsCXXRecordDecl();
+    if (!recordDecl)
+    {
+        return;
+    }
+
+    const auto *decl = dyn_cast<CXXRecordDecl>(recordDecl);
     if (!decl || !decl->getDefinition()
         || !(decl->getNameAsString() == "QVariant" || decl->getNameAsString() == "QHash" || decl->getNameAsString() == "QMap")) {
         return;
