diff --git a/src/plugins/projectexplorer/projectexplorer.cpp b/src/plugins/projectexplorer/projectexplorer.cpp
index cf82aa6e..9fa6050b 100644
--- a/src/plugins/projectexplorer/projectexplorer.cpp
+++ b/src/plugins/projectexplorer/projectexplorer.cpp
@@ -119,6 +119,7 @@
 
 #include <extensionsystem/pluginmanager.h>
 #include <extensionsystem/pluginspec.h>
+#include <extensionsystem/invoker.h>
 
 #include <texteditor/findinfiles.h>
 #include <texteditor/tabsettings.h>
@@ -2970,6 +2971,16 @@ void ProjectExplorerPlugin::runProject(Project *pro, Id mode, const bool forceSk
 {
     if (!pro)
         return;
+        
+    for (auto *rc : ProjectExplorerPlugin::instance()->allRunControls()) {
+        if (rc->isRunning() && rc->project() == pro && 
+            mode == ProjectExplorer::Constants::DEBUG_RUN_MODE) {
+            ExtensionSystem::Invoker<void>(
+                ExtensionSystem::PluginManager::getObjectByName("DebuggerPlugin"), 
+                "attachExternalApplication", rc);
+            return;
+        }
+    }
 
     if (Target *target = pro->activeTarget())
         if (RunConfiguration *rc = target->activeRunConfiguration())
