diff -cr a/share/qtcreator/translations/qtcreator_ru.ts b/share/qtcreator/translations/qtcreator_ru.ts
*** a/share/qtcreator/translations/qtcreator_ru.ts	2019-01-15 21:03:53.000000000 +0300
--- b/share/qtcreator/translations/qtcreator_ru.ts	2019-01-17 19:56:43.489942378 +0300
***************
*** 5400,5406 ****
      </message>
      <message>
          <source>&lt;Other Locations&gt;</source>
!         <translation>&lt;Другие места&gt;</translation>
      </message>
      <message>
          <source>CMake Project was parsed successfully.</source>
--- 5400,5406 ----
      </message>
      <message>
          <source>&lt;Other Locations&gt;</source>
!         <translation>Другие места</translation>
      </message>
      <message>
          <source>CMake Project was parsed successfully.</source>
***************
*** 5432,5438 ****
      </message>
      <message>
          <source>&lt;Headers&gt;</source>
!         <translation>&lt;Заголовки&gt;</translation>
      </message>
  </context>
  <context>
--- 5432,5438 ----
      </message>
      <message>
          <source>&lt;Headers&gt;</source>
!         <translation>Заголовки</translation>
      </message>
  </context>
  <context>
