diff -cr a/Telegram/SourceFiles/media/view/media_view_pip.cpp b/Telegram/SourceFiles/media/view/media_view_pip.cpp
*** a/Telegram/SourceFiles/media/view/media_view_pip.cpp	2022-06-21 01:04:35.000000000 +0300
--- b/Telegram/SourceFiles/media/view/media_view_pip.cpp	2022-06-22 19:36:43.454582552 +0300
***************
*** 475,481 ****
  	const auto parentScreen = widgetScreen(_parent);
  	const auto myScreen = widgetScreen(widget());
  	if (parentScreen && myScreen && myScreen != parentScreen) {
! 		widget()->setScreen(parentScreen);
  	}
  	auto position = Position();
  	position.snapped = RectPart::Top | RectPart::Left;
--- 475,481 ----
  	const auto parentScreen = widgetScreen(_parent);
  	const auto myScreen = widgetScreen(widget());
  	if (parentScreen && myScreen && myScreen != parentScreen) {
! 		widget()->windowHandle()->setScreen(parentScreen);
  	}
  	auto position = Position();
  	position.snapped = RectPart::Top | RectPart::Left;
