*** clementine-orig/src/ui/mainwindow.ui	2016-12-17 23:40:19.000000000 +0300
--- clementine-new/src/ui/mainwindow.ui	2016-12-17 23:41:52.539990467 +0300
***************
*** 57,63 ****
           <number>0</number>
          </property>
          <item>
!          <widget class="FancyTabWidget" name="tabs" native="true"/>
          </item>
          <item>
           <widget class="NowPlayingWidget" name="now_playing" native="true"/>
--- 57,70 ----
           <number>0</number>
          </property>
          <item>
!          <widget class="FancyTabWidget" name="tabs" native="true">
!           <property name="minimumSize">
!            <size>
!             <width>362</width>
!             <height>0</height>
!            </size>
!           </property>
!          </widget>
          </item>
          <item>
           <widget class="NowPlayingWidget" name="now_playing" native="true"/>
***************
*** 477,483 ****
       <x>0</x>
       <y>0</y>
       <width>1131</width>
!      <height>23</height>
      </rect>
     </property>
     <widget class="QMenu" name="menu_music">
--- 484,490 ----
       <x>0</x>
       <y>0</y>
       <width>1131</width>
!      <height>18</height>
      </rect>
     </property>
     <widget class="QMenu" name="menu_music">
