*** clementine-1.0.1/src/ui/mainwindow.cpp	2012-01-22 12:43:26.000000000 +0000
--- clementine-roand/src/ui/mainwindow.cpp.new	2012-08-08 22:57:03.803256098 +0000
***************
*** 275,281 ****
        device_view_container_,
        IconLoader::Load("multimedia-player-ipod-mini-blue", IconLoader::Base),
        tr("Devices"));
-   ui_->tabs->addSpacer();
    ui_->tabs->addTab(song_info_view_,
                      IconLoader::Load("view-media-lyrics", IconLoader::Base),
                      tr("Song info"));
--- 275,280 ----
***************
*** 1193,1199 ****
  }
  
  void MainWindow::SongChanged(const Song& song) {
-   setWindowTitle(song.PrettyTitleWithArtist());
    if (tray_icon_) tray_icon_->SetProgress(0);
  
  #ifdef HAVE_LIBLASTFM
--- 1192,1197 ----
