*** config.def.h.orig       2020-05-15 09:29:45.438319498 +0300
--- config.def.h    2020-05-15 09:30:41.997995602 +0300
***************
*** 8,13 ****
--- 8,18 ----
  static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
  static int borderpx = 2;
  
+ /* disable bold, italic and roman fonts globally */
+ int disablebold = 0;
+ int disableitalic = 0;
+ int disableroman = 0;
+ 
  /*
   * What program is execed by st depends of these precedence rules:
   * 1: program passed with -e
*** x.c.orig	2020-04-27 14:58:27.000000000 +0300
--- x.c	2020-05-15 09:27:21.747367604 +0300
***************
*** 254,259 ****
--- 254,264 ----
  
  static int oldbutton = 3; /* button event on startup: 3 = release */
  
+ /* declared in config.h */
+ extern int disablebold;
+ extern int disableitalic;
+ extern int disableroman;
+ 
  void
  clipcopy(const Arg *dummy)
  {
***************
*** 1002,1018 ****
  	win.ch = ceilf(dc.font.height * chscale);
  
  	FcPatternDel(pattern, FC_SLANT);
! 	FcPatternAddInteger(pattern, FC_SLANT, FC_SLANT_ITALIC);
  	if (xloadfont(&dc.ifont, pattern))
  		die("can't open font %s\n", fontstr);
  
  	FcPatternDel(pattern, FC_WEIGHT);
! 	FcPatternAddInteger(pattern, FC_WEIGHT, FC_WEIGHT_BOLD);
  	if (xloadfont(&dc.ibfont, pattern))
  		die("can't open font %s\n", fontstr);
  
  	FcPatternDel(pattern, FC_SLANT);
! 	FcPatternAddInteger(pattern, FC_SLANT, FC_SLANT_ROMAN);
  	if (xloadfont(&dc.bfont, pattern))
  		die("can't open font %s\n", fontstr);
  
--- 1007,1026 ----
  	win.ch = ceilf(dc.font.height * chscale);
  
  	FcPatternDel(pattern, FC_SLANT);
! 	if (!disableitalic)
! 	    FcPatternAddInteger(pattern, FC_SLANT, FC_SLANT_ITALIC);
  	if (xloadfont(&dc.ifont, pattern))
  		die("can't open font %s\n", fontstr);
  
  	FcPatternDel(pattern, FC_WEIGHT);
! 	if (!disablebold)
! 	    FcPatternAddInteger(pattern, FC_WEIGHT, FC_WEIGHT_BOLD);
  	if (xloadfont(&dc.ibfont, pattern))
  		die("can't open font %s\n", fontstr);
  
  	FcPatternDel(pattern, FC_SLANT);
! 	if (!disableroman)
! 	    FcPatternAddInteger(pattern, FC_SLANT, FC_SLANT_ROMAN);
  	if (xloadfont(&dc.bfont, pattern))
  		die("can't open font %s\n", fontstr);
  
