diff -cr i/tools/perf/util/annotate.c w/tools/perf/util/annotate.c
*** i/tools/perf/util/annotate.c  2022-08-30 14:23:00.576555268 +0300
--- w/tools/perf/util/annotate.c  2022-08-30 14:37:10.409875714 +0300
***************
*** 1721,1726 ****
--- 1721,1738 ----
  #include <bpf/libbpf.h>
  #include <linux/btf.h>
  
+ static int fprintf_styled(void *, enum disassembler_style, const char* fmt, ...)
+ {
+   va_list args;
+   int r;
+ 
+   va_start(args, fmt);
+   r = vprintf(fmt, args);
+   va_end(args);
+ 
+   return r;
+ }
+ 
  static int symbol__disassemble_bpf(struct symbol *sym,
  				   struct annotate_args *args)
  {
***************
*** 1763,1770 ****
  		goto out;
  	}
  	init_disassemble_info(&info, s,
! 			      (fprintf_ftype) fprintf);
! 
  	info.arch = bfd_get_arch(bfdf);
  	info.mach = bfd_get_mach(bfdf);
  
--- 1775,1782 ----
  		goto out;
  	}
  	init_disassemble_info(&info, s,
!                  (fprintf_ftype) fprintf,
!                  fprintf_styled);
  	info.arch = bfd_get_arch(bfdf);
  	info.mach = bfd_get_mach(bfdf);
  
