diff --git a/zaread b/zaread
index 65093d9d..f3305509 100755
--- a/zaread
+++ b/zaread
@@ -20,7 +20,7 @@ READER_ARG=""
 # Default converters
 MOBI_CMD="ebook-convert"
 OFFICE_CMD="soffice"
-MD_CMD="md2pdf"
+MD_CMD="md2pdf.sh"
 TYPST_CMD="typst"
 TYPST_ARG="compile"
 MD_ARG=""
@@ -253,7 +253,7 @@ case "$file_converter" in
 		;;
 	"$MD_CMD")
 		# shellcheck disable=SC2086
-		run_convert "$MD_CMD" "$MD_CMD" $MD_ARG "$directory/$file" -o "$ZA_CACHE_DIR/$pdffile.part"
+		run_convert "$MD_CMD" "$MD_CMD" $MD_ARG "$directory/$file" "$ZA_CACHE_DIR/$pdffile.part"
 		;;
 	"$TYPST_CMD")
 		# shellcheck disable=SC2086
