From 36eb13128974d4f7922cbb54c7da65c526ec9618 Mon Sep 17 00:00:00 2001
From: Jonas Kvinge <jonas@jkvinge.net>
Date: Sat, 21 May 2022 19:09:46 +0200
Subject: [PATCH] MainWindow: Remove QEvent::spontaneous() check in close event

Fixes #964
---
 src/core/mainwindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/mainwindow.cpp b/src/core/mainwindow.cpp
index 2ec491912..bbc1b32b4 100644
--- a/src/core/mainwindow.cpp
+++ b/src/core/mainwindow.cpp
@@ -1597,7 +1597,7 @@ void MainWindow::showEvent(QShowEvent *e) {
 void MainWindow::closeEvent(QCloseEvent *e) {
 
   if (!exit_) {
-    if (!hidden_ && keep_running_ && e->spontaneous() && tray_icon_->IsSystemTrayAvailable()) {
+    if (!hidden_ && keep_running_ && tray_icon_->IsSystemTrayAvailable()) {
       SetHiddenInTray(true);
     }
     else {
