diff --git a/src/systemtrayicon/qtsystemtrayicon.cpp b/src/systemtrayicon/qtsystemtrayicon.cpp
index 01822202..564a5ddb 100644
--- a/src/systemtrayicon/qtsystemtrayicon.cpp
+++ b/src/systemtrayicon/qtsystemtrayicon.cpp
@@ -194,6 +194,7 @@ void SystemTrayIcon::MuteButtonStateChanged(const bool value) {
 
 void SystemTrayIcon::SetNowPlaying(const Song &song, const QUrl &url) {
   Q_UNUSED(url)
+  return;
   if (available_) setToolTip(song.PrettyTitleWithArtist());
 }
 
