*** old/src/gqview.h	2006-11-01 20:28:37.000000000 +0300
--- new/src/gqview.h	2023-07-08 12:28:36.905561524 +0300
***************
*** 61,67 ****
   *----------------------------------------------------------------------------
   */
  
! #define GQVIEW_RC_DIR             ".gqview"
  #define GQVIEW_RC_DIR_COLLECTIONS GQVIEW_RC_DIR"/collections"
  #define GQVIEW_RC_DIR_TRASH       GQVIEW_RC_DIR"/trash"
  
--- 61,67 ----
   *----------------------------------------------------------------------------
   */
  
! #define GQVIEW_RC_DIR             ".config/gqview"
  #define GQVIEW_RC_DIR_COLLECTIONS GQVIEW_RC_DIR"/collections"
  #define GQVIEW_RC_DIR_TRASH       GQVIEW_RC_DIR"/trash"
  
*** old/src/print.c	2006-12-01 22:36:55.000000000 +0300
--- new/src/print.c	2023-07-08 12:25:51.183137583 +0300
***************
*** 1220,1226 ****
  		}
  	text[6] = '\0';
  
! 	fprintf(f, text);
  }                                                                                                                         
  static gint print_job_ps_page_image(PrintWindow *pw, GdkPixbuf *pixbuf,
  				    gdouble x, gdouble y, gdouble w, gdouble h,
--- 1220,1226 ----
  		}
  	text[6] = '\0';
  
! 	fprintf(f, "%s", text);
  }                                                                                                                         
  static gint print_job_ps_page_image(PrintWindow *pw, GdkPixbuf *pixbuf,
  				    gdouble x, gdouble y, gdouble w, gdouble h,
***************
*** 1345,1351 ****
  		text[1] = hex_digits[*p & 0xf];
  		text[2] = '\0';
  
! 		fprintf(f, text);
  
  		p++;
  		}
--- 1345,1351 ----
  		text[1] = hex_digits[*p & 0xf];
  		text[2] = '\0';
  
! 		fprintf(f,"%s", text);
  
  		p++;
  		}
*** old/src/ui_fileops.c	2006-10-13 18:32:31.000000000 +0400
--- new/src/ui_fileops.c	2023-07-08 12:26:33.309573083 +0300
***************
*** 44,50 ****
  	gchar *text_l;
  
  	text_l = g_locale_from_utf8(text_utf8, -1, NULL, NULL, NULL);
! 	printf((text_l) ? text_l : text_utf8);
  	g_free(text_l);
  }
  
--- 44,50 ----
  	gchar *text_l;
  
  	text_l = g_locale_from_utf8(text_utf8, -1, NULL, NULL, NULL);
! 	printf("%s", (text_l) ? text_l : text_utf8);
  	g_free(text_l);
  }
  
*** old/src/view_file_icon.c	2006-12-01 01:55:00.000000000 +0300
--- new/src/view_file_icon.c	2023-07-08 12:24:43.126844372 +0300
***************
*** 614,620 ****
  	uri_text = uri_text_from_list(list, &total, (info == TARGET_TEXT_PLAIN));
  	path_list_free(list);
  
- 	if (debug) printf(uri_text);
  
  	gtk_selection_data_set(selection_data, selection_data->target,
  			       8, (guchar *)uri_text, total);
--- 614,619 ----
*** old/src/view_file_list.c	2006-12-01 01:55:00.000000000 +0300
--- new/src/view_file_list.c	2023-07-08 12:24:46.396826428 +0300
***************
*** 136,142 ****
  	uri_text = uri_text_from_list(list, &total, (info == TARGET_TEXT_PLAIN));
  	path_list_free(list);
  
- 	if (debug) printf(uri_text);
  
  	gtk_selection_data_set(selection_data, selection_data->target,
  			       8, (guchar *)uri_text, total);
--- 136,141 ----
*** old/src/thumb_standard.h   2005-01-06 04:08:55.000000000 +0300
--- new/src/thumb_standard.h   2023-07-21 21:25:14.700015048 +0300
***************
*** 14,20 ****
  #define THUMB_STANDARD_H
  
  
! #define THUMB_FOLDER        ".thumbnails"
  #define THUMB_FOLDER_NORMAL "normal"
  #define THUMB_FOLDER_LARGE  "large"
  #define THUMB_FOLDER_FAIL   "fail/gqview-1.0"
--- 14,20 ----
  #define THUMB_STANDARD_H
  
  
! #define THUMB_FOLDER        ".config/gqview/thumb_folder"
  #define THUMB_FOLDER_NORMAL "normal"
  #define THUMB_FOLDER_LARGE  "large"
  #define THUMB_FOLDER_FAIL   "fail/gqview-1.0"
