diff --git a/tracopt/versioncontrol/git/git_fs.py b/tracopt/versioncontrol/git/git_fs.py
index 77bc95013..333324303 100644
Index: tracopt/versioncontrol/git/git_fs.py
===================================================================
--- a/tracopt/versioncontrol/git/git_fs.py
+++ b/tracopt/versioncontrol/git/git_fs.py
@@ -645,6 +645,7 @@ class GitNode(Node):
             rev = repos.normalize_rev(to_unicode(rev))
         else:
             rev = repos.youngest_rev
+        created_rev = rev
 
         kind = Node.DIRECTORY
         p = path.strip('/')
@@ -662,7 +663,7 @@ class GitNode(Node):
             self.fs_perm, k, self.fs_sha, self.fs_size, fname = ls_tree_info
 
             # fix-up to the last commit-rev that touched this node
-            rev = repos.git.last_change(rev, p, historian)
+            created_rev = repos.git.last_change(rev, p, historian)
 
             if k == 'tree':
                 pass
@@ -677,7 +678,7 @@ class GitNode(Node):
                                   "kind '%(kind)s')", kind=k))
 
         self.created_path = path
-        self.created_rev = rev
+        self.created_rev = created_rev
 
         Node.__init__(self, repos, path, rev, kind)
 