*** a/Documentation/sphinx/kernel_include.py   2024-05-28 12:01:23.983386800 +0300
--- a/Documentation/sphinx/kernel_include.py   2024-05-28 12:04:03.576728097 +0300
***************
*** 93,98 ****
--- 93,99 ----
          if path.startswith('<') and path.endswith('>'):
              path = os.path.join(self.standard_include_path, path[1:-1])
          path = os.path.normpath(os.path.join(source_dir, path))
+         path = str(path)
  
          # HINT: this is the only line I had to change / commented out:
          #path = utils.relative_path(None, path)

