*** old/chrome/browser/ui/layout_constants.cc	2019-02-07 01:06:38.000000000 +0300
--- new/chrome/browser/ui/layout_constants.cc	2019-02-11 00:59:36.133485914 +0300
***************
*** 44,50 ****
      case HOSTED_APP_PAGE_ACTION_ICON_SIZE:
        // We must limit the size of icons in the title bar to avoid vertically
        // stretching the container view.
!       return 16;
      case LOCATION_BAR_BUBBLE_FONT_VERTICAL_PADDING:
        return 2;
      case LOCATION_BAR_BUBBLE_CORNER_RADIUS:
--- 44,50 ----
      case HOSTED_APP_PAGE_ACTION_ICON_SIZE:
        // We must limit the size of icons in the title bar to avoid vertically
        // stretching the container view.
!       return 0;
      case LOCATION_BAR_BUBBLE_FONT_VERTICAL_PADDING:
        return 2;
      case LOCATION_BAR_BUBBLE_CORNER_RADIUS:
***************
*** 63,79 ****
      case LOCATION_BAR_ICON_SIZE:
        return touch_ui ? 20 : 16;
      case TAB_AFTER_TITLE_PADDING:
!       return touch_ui ? 8 : 4;
      case TAB_ALERT_INDICATOR_CAPTURE_ICON_WIDTH:
!       return 16;
      case TAB_ALERT_INDICATOR_ICON_WIDTH:
!       return touch_ui ? 12 : 16;
      case TAB_HEIGHT:
!       return (touch_ui ? 41 : 34) + GetLayoutConstant(TABSTRIP_TOOLBAR_OVERLAP);
      case TAB_PRE_TITLE_PADDING:
        return 8;
      case TAB_STACK_DISTANCE:
!       return touch_ui ? 4 : 6;
      case TABSTRIP_TOOLBAR_OVERLAP:
        return 1;
      case TOOLBAR_ELEMENT_PADDING:
--- 63,79 ----
      case LOCATION_BAR_ICON_SIZE:
        return touch_ui ? 20 : 16;
      case TAB_AFTER_TITLE_PADDING:
!       return touch_ui ? 8 : 0;
      case TAB_ALERT_INDICATOR_CAPTURE_ICON_WIDTH:
!       return 0;
      case TAB_ALERT_INDICATOR_ICON_WIDTH:
!       return touch_ui ? 12 : 0;
      case TAB_HEIGHT:
!       return (touch_ui ? 24 : 24) + GetLayoutConstant(TABSTRIP_TOOLBAR_OVERLAP);
      case TAB_PRE_TITLE_PADDING:
        return 8;
      case TAB_STACK_DISTANCE:
!       return touch_ui ? 4 : 0;
      case TABSTRIP_TOOLBAR_OVERLAP:
        return 1;
      case TOOLBAR_ELEMENT_PADDING:
*** old/chrome/browser/ui/views/chrome_typography.cc	2019-02-07 01:06:38.000000000 +0300
--- new/chrome/browser/ui/views/chrome_typography.cc	2019-02-09 13:55:13.339947057 +0300
***************
*** 75,81 ****
        const int omnibox_primary_delta =
            GetFontSizeDeltaBoundedByAvailableHeight(
                LocationBarView::GetAvailableTextHeight(),
!               ui::TouchUiController::Get()->touch_ui() ? 15 : 14);
        *size_delta = omnibox_primary_delta;
        if (context == CONTEXT_OMNIBOX_DEEMPHASIZED) {
          (*size_delta)--;
--- 75,81 ----
        const int omnibox_primary_delta =
            GetFontSizeDeltaBoundedByAvailableHeight(
                LocationBarView::GetAvailableTextHeight(),
!               ui::TouchUiController::Get()->touch_ui() ? 13 : 13);
        *size_delta = omnibox_primary_delta;
        if (context == CONTEXT_OMNIBOX_DEEMPHASIZED) {
          (*size_delta)--;
*** old/chrome/browser/ui/views/tabs/new_tab_button.cc	2019-02-07 01:06:38.000000000 +0300
--- new/chrome/browser/ui/views/tabs/new_tab_button.cc	2019-02-10 12:05:03.598842169 +0300
***************
*** 58,64 ****
  }  // namespace
  
  // static
! const gfx::Size NewTabButton::kButtonSize{28, 28};
  
  NewTabButton::NewTabButton(TabStrip* tab_strip, views::ButtonListener* listener)
      : views::ImageButton(listener), tab_strip_(tab_strip) {
--- 58,64 ----
  }  // namespace
  
  // static
! const gfx::Size NewTabButton::kButtonSize{18, 18};
  
  NewTabButton::NewTabButton(TabStrip* tab_strip, views::ButtonListener* listener)
      : views::ImageButton(listener), tab_strip_(tab_strip) {
*** old/chrome/browser/ui/views/tabs/tab_strip.cc	2019-02-07 01:06:38.000000000 +0300
--- new/chrome/browser/ui/views/tabs/tab_strip.cc	2019-02-10 12:17:46.623499794 +0300
***************
*** 3557,3563 ****
    if (tab_search_button_)
      tab_search_button_->SetBorder(views::CreateEmptyBorder(button_insets));
  
!   constexpr int kHorizontalInset = 8;
    tab_controls_container_->SetBorder(views::CreateEmptyBorder(
        gfx::Insets(0, kHorizontalInset, 0, kHorizontalInset)));
  }
--- 3557,3563 ----
    if (tab_search_button_)
      tab_search_button_->SetBorder(views::CreateEmptyBorder(button_insets));
  
!   constexpr int kHorizontalInset = 3;
    tab_controls_container_->SetBorder(views::CreateEmptyBorder(
        gfx::Insets(0, kHorizontalInset, 0, kHorizontalInset)));
  }
