diff -cr a/qt/IconCache.cc b/qt/IconCache.cc
*** a/qt/IconCache.cc	2023-02-22 22:12:15.000000000 +0300
--- b/qt/IconCache.cc	2023-03-25 10:02:28.374231415 +0300
***************
*** 98,126 ****
  
          return icon;
      }
! 
!     auto const mime_icon = getMimeTypeIcon(mime_type_name, false);
!     for (auto const& size : { QSize(24, 24), QSize(32, 32), QSize(48, 48) })
!     {
!         // upper left corner
!         auto const folder_size = size / 2;
!         auto const folder_rect = QRect(QPoint(), folder_size);
! 
!         // fullsize
!         auto const mime_size = size;
!         auto const mime_rect = QRect(QPoint(), mime_size);
! 
!         // build the icon
!         auto pixmap = QPixmap(size);
!         pixmap.fill(Qt::transparent);
!         QPainter painter(&pixmap);
!         painter.setRenderHints(QPainter::SmoothPixmapTransform);
!         painter.drawPixmap(folder_rect, folder_icon_.pixmap(folder_size));
!         painter.drawPixmap(mime_rect, mime_icon.pixmap(mime_size));
!         icon.addPixmap(pixmap);
!     }
! 
!     return icon;
  }
  
  QIcon IconCache::getThemeIcon(QString const& name, std::optional<QStyle::StandardPixmap> const& fallback) const
--- 98,105 ----
  
          return icon;
      }
!     
!     return folder_icon_;
  }
  
  QIcon IconCache::getThemeIcon(QString const& name, std::optional<QStyle::StandardPixmap> const& fallback) const
diff -cr a/qt/Application.cc b/qt/Application.cc
*** a/qt/Application.cc	2023-02-22 22:12:15.000000000 +0300
--- b/qt/Application.cc	2023-03-25 09:32:10.190853355 +0300
***************
*** 87,93 ****
      }
  
      // if that fails, use our own as the fallback
!     return QIcon{ QStringLiteral(":/icons/transmission.svg") };
  }
  
  } // namespace
--- 87,93 ----
      }
  
      // if that fails, use our own as the fallback
!     return QIcon{ QStringLiteral(":/icons/transmission.png") };
  }
  
  } // namespace
diff -cr a/qt/application.qrc b/qt/application.qrc
*** a/qt/application.qrc	2023-02-22 22:12:15.000000000 +0300
--- b/qt/application.qrc	2023-03-25 09:32:00.100853105 +0300
***************
*** 5,10 ****
        <file alias="encrypted.svg">icons/lock.svg</file>
        <file alias="hamburger-menu.svg">icons/hamburger-menu.svg</file>
        <file alias="ratio.svg">icons/yin-yang.svg</file>
!       <file alias="transmission.svg">icons/transmission.svg</file>
    </qresource>
  </RCC>
--- 5,10 ----
        <file alias="encrypted.svg">icons/lock.svg</file>
        <file alias="hamburger-menu.svg">icons/hamburger-menu.svg</file>
        <file alias="ratio.svg">icons/yin-yang.svg</file>
!       <file alias="transmission.png">icons/transmission.png</file>
    </qresource>
  </RCC>

diff -cr old/qt/MainWindow.ui new/qt/MainWindow.ui
*** old/qt/MainWindow.ui	2020-05-22 14:04:23.456805346 +0300
--- new/qt/MainWindow.ui	2020-05-23 13:43:50.759382125 +0300
***************
*** 21,27 ****
      <property name="spacing">
       <number>0</number>
      </property>
!     <property name="margin">
       <number>0</number>
      </property>
      <item>
--- 21,36 ----
      <property name="spacing">
       <number>0</number>
      </property>
!      <property name="leftMargin">
!       <number>0</number>
!      </property>
!      <property name="topMargin">
!       <number>0</number>
!      </property>
!      <property name="rightMargin">
!       <number>0</number>
!      </property>
!      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
***************
*** 303,309 ****
      <bool>false</bool>
     </attribute>
     <addaction name="action_OpenFile"/>
-    <addaction name="action_AddURL"/>
     <addaction name="action_Start"/>
     <addaction name="action_Pause"/>
     <addaction name="action_Remove"/>
--- 312,317 ----
diff -cr a/qt/StyleHelper.cc b/qt/StyleHelper.cc
*** a/qt/StyleHelper.cc	2023-02-22 22:12:15.000000000 +0300
--- b/qt/StyleHelper.cc	2023-03-25 09:17:13.364164620 +0300
***************
*** 5,10 ****
--- 5,11 ----
  
  #include <QPainter>
  #include <QStyleOptionProgressBar>
+ #include <QStyleFactory>
  
  #include "StyleHelper.h"
  
***************
*** 36,41 ****
      style.drawControl(QStyle::CE_ProgressBar, &my_option, &painter);
      painter.restore();
  #else
!     style.drawControl(QStyle::CE_ProgressBar, &option, &painter);
  #endif
  }
--- 37,42 ----
      style.drawControl(QStyle::CE_ProgressBar, &my_option, &painter);
      painter.restore();
  #else
!     QStyleFactory::create(QString::fromLatin1 ("Fusion"))->drawControl(QStyle::CE_ProgressBar, &option, &painter);
  #endif
  }
diff -cr a/qt/TorrentDelegate.cc b/qt/TorrentDelegate.cc
*** a/qt/TorrentDelegate.cc	2023-02-22 22:12:15.000000000 +0300
--- b/qt/TorrentDelegate.cc	2023-03-25 09:20:42.924169772 +0300
***************
*** 302,308 ****
      case TR_STATUS_DOWNLOAD:
      case TR_STATUS_SEED:
          str = shortTransferString(tor) + QStringLiteral("    ") +
!             tr("Ratio: %1").arg(Formatter::get().ratioToString(tor.ratio()));
          break;
  
      default:
--- 302,308 ----
      case TR_STATUS_DOWNLOAD:
      case TR_STATUS_SEED:
          str = shortTransferString(tor) + QStringLiteral("    ") +
!             tr("%1").arg(Formatter::get().ratioToString(tor.ratio()));
          break;
  
      default:
diff -cr a/qt/TorrentDelegateMin.cc b/qt/TorrentDelegateMin.cc
*** a/qt/TorrentDelegateMin.cc	2023-02-22 22:12:15.000000000 +0300
--- b/qt/TorrentDelegateMin.cc	2023-03-25 09:22:02.987505067 +0300
***************
*** 30,36 ****
  enum
  {
      GUI_PAD = 6,
!     BAR_WIDTH = 50,
      BAR_HEIGHT = 16,
      LINE_SPACING = 4
  };
--- 30,36 ----
  enum
  {
      GUI_PAD = 6,
!     BAR_WIDTH = 45,
      BAR_HEIGHT = 16,
      LINE_SPACING = 4
  };
***************
*** 111,117 ****
      auto const name_fm = QFontMetrics(name_font);
      auto const name_size = name_fm.size(0, name_text_);
  
-     status_font.setPointSize(static_cast<int>(status_font.pointSize() * 0.85));
      QFontMetrics const status_fm(status_font);
      QSize const status_size(status_fm.size(0, status_text_));
  
--- 111,116 ----
