diff --git a/src/Parsers/CppParser/cppdocumentparser.cpp b/src/Parsers/CppParser/cppdocumentparser.cpp
index 60bdb47e..76c7f205 100644
--- a/src/Parsers/CppParser/cppdocumentparser.cpp
+++ b/src/Parsers/CppParser/cppdocumentparser.cpp
@@ -37,7 +37,6 @@
 #include <cppeditor/cppmodelmanager.h>
 #include <cppeditor/cpptoolsreuse.h>
 #include <projectexplorer/project.h>
-#include <texteditor/syntaxhighlighter.h>
 #include <texteditor/texteditor.h>
 #include <utils/algorithm.h>
 #include <utils/mimeutils.h>
diff --git a/src/spellcheckquickfix.cpp b/src/spellcheckquickfix.cpp
index 6f0bf6c0..9c47c8e6 100644
--- a/src/spellcheckquickfix.cpp
+++ b/src/spellcheckquickfix.cpp
@@ -94,7 +94,7 @@ public:
 
 } // namespace Internal
 
-void SpellCheckCppQuickFixFactory::match( const CppEditor::Internal::CppQuickFixInterface& interface, TextEditor::QuickFixOperations& result )
+void SpellCheckCppQuickFixFactory::doMatch( const CppEditor::Internal::CppQuickFixInterface& interface, TextEditor::QuickFixOperations& result )
 {
   Q_UNUSED( interface )
 
diff --git a/src/spellcheckquickfix.h b/src/spellcheckquickfix.h
index 1c77320b..8d4b5030 100644
--- a/src/spellcheckquickfix.h
+++ b/src/spellcheckquickfix.h
@@ -33,7 +33,7 @@ class SpellCheckCppQuickFixFactory
   : public CppEditor::CppQuickFixFactory
 {
 public:
-  void match( const CppEditor::Internal::CppQuickFixInterface& interface, TextEditor::QuickFixOperations& result ) Q_DECL_OVERRIDE;
+  void doMatch( const CppEditor::Internal::CppQuickFixInterface& interface, TextEditor::QuickFixOperations& result ) Q_DECL_OVERRIDE;
 };
 
 } // namespace SpellChecker
